runtime/gc_structs/ClassArrayClassSlotIterator.cpp

```diff
@@ -50,6 +50,16 @@ GC_ClassArrayClassSlotIterator::nextSlot()
 				_state += 1;
 			}
 			break;
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+		case classArrayClassSlotIterator_state_nullRestrictedArrayClass:
```

Comment:
IIRC should this state do the following

```
		if (!_isArrayClass) {
				_state = classArrayClassSlotIterator_state_done;
			} else {
```

Given that a non-array class now potentially points to two array classes?

Reply:
Well, I think not here at least. This iterator is one of collection of iterators that `GC_ClassIteratorClassSlots` calls. This particular one suppose to return extra specific slots if given class is an array class. So, this check means "skip is given class is not an array class". If class is not an array class this iterator should return nothing in my opinion. However there is a question should any other iterator from this collection return new slots?

Reply:
If _IsArrayClass is true there will never be a nullRestrictedArrayClass, so maybe this case isn't needed at all.

Reply:
Okay, I understand. Its fine as is.

Reply:
My previous comment was in response to Dmitri

> If _IsArrayClass is true there will never be a nullRestrictedArrayClass, so maybe this case isn't needed at all.

Yes, since `companionArray` and `nullRestrictedArrayClass` are the same slot then we only need one which can be `companionArray`

Reply:
I removed the nullRestrictedArrayClass case.